compilers/concrete-optimizer/concrete-optimizer/src/optimization/dag/multi_parameters/analyze.rs

```diff
@@ -61,10 +62,34 @@ pub fn analyze(
         Some(p_cut) => p_cut.clone(),
         None => maximal_p_cut(&dag),
     };
-    let partitions = partitionning_with_preferred(&dag, &p_cut, default_partition);
+    let partitions =
+        partitionning_with_preferred(&dag, &p_cut, default_partition, allow_composition);
     let instrs_partition = partitions.instrs_partition;
     let nb_partitions = partitions.nb_partitions;
-    let out_variances = out_variances(&dag, nb_partitions, &instrs_partition);
+    let mut out_variances = self::out_variances(&dag, nb_partitions, &instrs_partition, &None);
+    if allow_composition {
+        // Verify that there is no input symbol in the symbolic variances of the outputs.
+        assert_no_input_var_in_out_var(&dag, &out_variances, nb_partitions);
```

Comment:
should be an error here instead of an assert

Reply:
maybe a field could be added,
like support_composition=true/false so the error is not part of the analysis